landa/water_body_management/doctype/catch_log_entry/catch_log_entry.py

```diff
@@ -20,3 +20,21 @@ def before_insert(self):
 			self.origin_of_catch_log_entry = "Regionalverband"
 		else:
 			self.origin_of_catch_log_entry = "Verein"
+
+	def validate(self):
+		tolerance = 0.4
+		for row in self.fish_catches:
+			species = row.fish_species   # wurde dreimal aus der Datenbank geholt deswegen hier das auslesen in eine extra variable
+			if frappe.db.get_value("Fish Species", species, "typical_weight"):  								 # check if the weight is available
+				typical_weight = float(frappe.db.get_value("Fish Species", species, "typical_weight"))
+				average_weight = row.weight_in_kg / row.amount
+				if (
+					average_weight < typical_weight * (1 - tolerance) or
+					average_weight > typical_weight * (1 + tolerance)
+				):
+					frappe.msgprint(
+						frappe._("The weight of {0} in row {1} is not plausible").format(species, row.idx)
+					)
+			else:
+				pass
+				#automatisierte E-Mail an den Verantwortlichen das Gewicht nach zu pflegen?
```

Comment:
```suggestion
			typical_weight = flt(frappe.db.get_value("Fish Species", row.fish_species, "typical_weight_in_kg"))
			average_weight = row.weight_in_kg / row.amount
			if (
				typical_weight and
				average_weight < typical_weight * (1 - tolerance) or
				average_weight > typical_weight * (1 + tolerance)
			):
				frappe.msgprint(
					frappe._("The weight of {0} in row {1} is not plausible").format(row.fish_species, row.idx)
				)
```

- `frappe.utils.flt()` verwandelt `None` zu `0.0`, `float()` wirft einen Fehler
- `typical_weight` nicht zweimal aus der Datenbank laden
- `row.fish_species` ist kein Datenbankabruf. Daher bringt es nicht viel, das in eine Variable zu packen
- `else: pass` ist das Standardverhalten und muss nicht explizit aufgeschrieben werden